### PR TITLE
Updating from server issue related to minimum stability (ref #17956)

### PR DESCRIPTION
**Pull request by ChristopheSeg**
Pull Request for Issue #17956 .

### Summary of Changes
add 4 flags in Jupdate object in order to save package name and to detect the cause of invalid extension update message
	$this->noPhpMatch = true;
	$this->noStabilityMatch = true;
	$this->noDbMatch = true;
	$this->packageName 

in update function replace _JText('COM_INSTALLER_INVALID_EXTENSION_UPDATE') by messages explaining the possible cause(s) of the "invalid extension update"

in language file modify/add corresponding language strings

### Testing Instructions

see https://issues.joomla.org/tracker/joomla-cms/17956

### Actual result
now, when upgrading an extension with insufficient PHP or Database or minimumm stability, the possible cause of the error is reported  

### Documentation Changes Required
none

### Additional comment
In case of invalid extension update, the name of the package is given. 
I supposed that one upgrade extension's manifest file only contains one package or one component. If several components or package (with different names) can be included in one manifest file the package name is no longer relevant.

### CODE REVIEW

**Comment by zero-24** [comment]
Hi @ChristopheSeg 

Thanks fo your PR. Can you take a look into the errors thrown by PHPCS http://ci.joomla.org/joomla/joomla-cms/25 ?

IMO when we start adding $this->no<whatever> why not convert the other variables containing the same information in that file too?

**Comment by ChristopheSeg** [comment]
Le 26/09/2017 23:08, zero-24 a écrit :
    
    
      Hi @ChristopheSeg
      Thanks fo your PR. Can you take a look into the errors thrown
        by PHPCS http://ci.joomla.org/joomla/joomla-cms/25
        ?
    
    Sorry I did not looked at coding standard: will modify this tomorrow
    
      IMO when we start adding $this->no why not convert the other
        variables containing the same information in that file too?
    
    the JUpdate object ($this ) added parameter
    $this->noStabilityMatch and the others do not carry the same
    information as $StabilityMatch
    $this->noStabilityMatch$this->noStabilityMatch and the other 
    
      —
        You are receiving this because you were mentioned.
        Reply to this email directly, view
          it on GitHub, or mute
          the thread.
      
        
          
          
        
        
      
      {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/joomla/joomla-cms","title":"joomla/joomla-cms","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/joomla/joomla-cms"}},"updates":{"snippets":[{"icon":"PERSON","message":"@zero-24 in #18132: Hi @ChristopheSeg \r\n\r\nThanks fo your PR. Can you take a look into the errors thrown by PHPCS http://ci.joomla.org/joomla/joomla-cms/25 ?\r\n\r\nIMO when we start adding $this-\u003eno\u003cwhatever\u003e why not convert the other variables containing the same information in that file too?"}],"action":{"name":"View Pull Request","ur
l":"https://github.com/joomla/joomla-cms/pull/18132#issuecomment-332336846"}}}

**Comment by ChristopheSeg** [comment]
Le 26/09/2017 23:08, zero-24 a écrit :
    
    
      Hi @ChristopheSeg
      Thanks fo your PR. Can you take a look into the errors thrown
        by PHPCS http://ci.joomla.org/joomla/joomla-cms/25
        ?
    
    Sorry I did not looked at coding standard: will modify this tomorrow
    
    
      IMO when we start adding $this->no why not convert the other
        variables containing the same information in that file too?
    
    Just before  the "invalid extension update" message is fired,
    (before returning from "$update->get('downloadurl')->_data)"
    the part of JUpdate object containing update information and name is
    unset: 
      unset($this->latest);
      unset($this->currentUpdate);
    and the xml manifest files is closed
    
    the JUpdate object ($this ) added parameter
    ($this->noStabilityMatch and the others) do not carry the same
    information as $StabilityMatch
    $StabilityMatch=true indicates that the current update (one among
    all in the update manifest file) match stability. This variable is
    calculated of each individual <update>...</update>
    contained in the manifest file
    $this->noStabilityMatch=true  indicates that none of these file
    match stability
    
    
    
      —
        You are receiving this because you were mentioned.
        Reply to this email directly, view
          it on GitHub, or mute
          the thread.
      
        
          
          
        
        
      
      {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/joomla/joomla-cms","title":"joomla/joomla-cms","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/joomla/joomla-cms"}},"updates":{"snippets":[{"icon":"PERSON","message":"@zero-24 in #18132: Hi @ChristopheSeg \r\n\r\nThanks fo your PR. Can you take a look into the errors thrown by PHPCS http://ci.joomla.org/joomla/joomla-cms/25 ?\r\n\r\nIMO when we start adding $this-\u003eno\u003cwhatever\u003e why not convert the other variables containing the same information in that file too?"}],"action":{"name":"View Pull Request","ur
l":"https://github.com/joomla/joomla-cms/pull/18132#issuecomment-332336846"}}}

File: administrator/language/en-GB/en-GB.com_installer.ini [hunk 1]
```diff
@@ -75,7 +75,10 @@ COM_INSTALLER_INSTALL_FROM_WEB_TOS="By selecting "_QQ_"Add Install from Web tab"
 COM_INSTALLER_INSTALL_LANGUAGE_SUCCESS="Installation of the <strong>%s</strong> language was successful."
 COM_INSTALLER_INSTALL_SUCCESS="Installation of the %s was successful."
 COM_INSTALLER_INSTALL_URL="Install URL"
-COM_INSTALLER_INVALID_EXTENSION_UPDATE="Invalid extension update"
+COM_INSTALLER_INVALID_EXTENSION_UPDATE="%s : Invalid extension update"
+COM_INSTALLER_EXTENSION_UPDATE_NO_PHP_MATCH="No file in %s upgrade match the website PHP requirements"
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.com_installer.ini]: Move/sort 3 keys in alpha order.

File: libraries/src/Updater/Update.php [hunk 2]
```diff
@@ -305,6 +309,12 @@ public function _endElement($parser, $name)
 		{
 			// Closing update, find the latest version and check
 			case 'UPDATE':
+				// Retrieve and save type and name of package from xml update file for invalid extension error message
+				if (! isset($this->packageName))
```
Comment (Quy) [inline libraries/src/Updater/Update.php]: Remove space after `!`

File: administrator/language/en-GB/en-GB.com_installer.ini [hunk 3]
```diff
@@ -26,6 +26,9 @@ COM_INSTALLER_EXTENSION_PACKAGE_FILE="Extension package file"
 COM_INSTALLER_EXTENSION_PROTECTED="Protected extension"
 COM_INSTALLER_EXTENSION_PUBLISHED="Extension enabled."
 COM_INSTALLER_EXTENSION_UNPUBLISHED="Extension disabled."
+COM_INSTALLER_EXTENSION_UPDATE_NO_PHP_MATCH="No file in %s upgrade match the website PHP requirements"
+COM_INSTALLER_EXTENSION_UPDATE_NO_STABILITY_MATCH="No file in %s upgrade match the website Minimum Stability requirements"
+COM_INSTALLER_EXTENSION_UPDATE_NO_DB_MATCH="No file in %s upgrade match the website Database requirements"
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.com_installer.ini]: Please move line 31 above line 29. Also, add ending period to these strings.

**Review by mbabker — APPROVED** [review]

**Comment by zero-24** [comment]
Dear @ChristopheSeg 

in preperation of the upcomming release of Joomla 3.10 we have used GitHubs rename feature to rename the staging branch into 3.10-dev. Usually GitHub moves all existing PRs towards the new branch just fine, but here it didnt work. The reason seems to be that the fork of the CMS that was used as base for this PR has been deleted so GitHub does no longer have a base to rebase the PR against the new branch and we are also not able to reopen the PR. For that reason GitHub closed this PR in my name, when this issue is still valid It would require a new PR against the new 3.10-dev or 4.0-dev branch.

**Pull request closed without merging**